test/parallel/test-http2-many-writes-and-destroy.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const http2 = require('http2');
+const assert = require('assert');
+
+{
+  const server = http2.createServer((req, res) => {
+    req.pipe(res);
+  });
+
+  setTimeout(() => {
+    assert(false, 'should exit before 10s');
+  }, 10000).unref();
```

Comment:
I think we've been removing these kinds of checks to avoid timeout-related flakiness in tests -- the test runner enforces its own timeout anyway. (/cc @trott)

If you want to have actual timing guarantees, you can try to remove the dependency on networking and use something like `test/parallel/test-http2-generic-streams.js` does.

Reply:
Other things to be aware of, just sorta FYI, you don't need to do anything with these at this time unless you think you should:

* Tests with timers like this can become unreliable if the test is competing with other tests for resources. This unreliability is introduced surprisingly often on a few platforms in CI (particularly FreeBSD, I *think* because `test.py` decides it has 48 processors or something but they're not "real" processors--never really dug into it TBH, but you get the idea). To avoid the problem, the test can be put in `sequential` rather than `parallel`.

* If you think 10 seconds is plenty for most things, but that maybe a Raspberry Pi could use a little more time than that, use `common.platformTimeout()`.

Reply:
:+1: